### DISCUSSION TITLE
--isolate or --careful, what should I choose?

### DISCUSSION THREAD

**Discussion by m-rq** [ablab/spades#1074] [Q&A]
I want to assemble the yeast genome, about 11MB-12.5MB. I used PE150 method for sequencing, with high coverage, the coverage ＞100X. After removing the adapters and filtering low quality base, each read is less than 150bp. The manual indicated that "--isolate flag is highly recommended for high-coverage isolate." At the same time，the article "BENCHMARKING TOOLS FOR DE NOVO MICROBIAL ASSEMBLY" indicated --careful. I noticed that "--isolate option is not compatible with --only-error-correction or --careful options." Only one of these two options can be selected. What should I choose?

In addition, after filtering pair reads (only one library), an unpaired reads file is generated. How should I set input?
`-1 <file_name> -2 <file_name> -s <file_name> `
or
`--pe1-1 <file_name> --pe1-2 <file_name> --pe1-s <file_name>`

After reading the article "BENCHMARKING TOOLS FOR DE NOVO MICROBIAL ASSEMBLY", I got confused. Should I add an option --only-assembler?

Thanks.

**Comment by asl** [comment]
I would suggest you to go with `--isolate` it's more recent option tuned for your case.

> Should I add an option --only-assembler?

It is already implied by `--isolate`, no additional options will be needed.

> In addition, after filtering pair reads (only one library), an unpaired reads file is generated. How should I set input?

Please refer to SPAdes manual: https://cab.spbu.ru/files/release3.15.5/manual.html#sec3.2 both ways are possible.

**Reply by maruiqi0710** [reply]
Before using -- isolate, do I need to use --only-error-correction or other NGS reads error correction tools?

**Reply by asl** [reply]
Well, if this is disabled in `--isolate` then it was done for a reason. We do not recommend to perform any read error correction (besides the usual adapter removal and light trimming)

**Reply by jvollme** [reply]
I am not exactly convinced that the "--isolate" settings are really the best options to use, even for assumed isolates. Because without an error correction, i am assuming a coverage cutoff is employed to minimize the effect of sequencing errors? Wouldn't that cause the --isolate mode to overlook possible low abundant residual contaminants in assumed "isolates"? 
I am asking, because it happens just too often that people simply assume they have pure cultures and may then think their assembly results after sequencing are sufficient proof of purity. For this reason I actually keep asking people NOT to use the --isolate option, ever. Or am I misunderstanding something here?
